privval/priv_validator.go

```diff
@@ -37,98 +37,151 @@ func voteToStep(vote *types.Vote) int8 {
 
 // FilePV implements PrivValidator using data persisted to disk
 // to prevent double signing.
-// NOTE: the directory containing the pv.filePath must already exist.
+// NOTE: the directories containing pv.Key.filePath and pv.LastSignState.filePath must already exist.
 // It includes the LastSignature and LastSignBytes so we don't lose the signature
 // if the process crashes after signing but before the resulting consensus message is processed.
 type FilePV struct {
-	Address       types.Address  `json:"address"`
-	PubKey        crypto.PubKey  `json:"pub_key"`
-	LastHeight    int64          `json:"last_height"`
-	LastRound     int            `json:"last_round"`
-	LastStep      int8           `json:"last_step"`
-	LastSignature []byte         `json:"last_signature,omitempty"`
-	LastSignBytes cmn.HexBytes   `json:"last_signbytes,omitempty"`
-	PrivKey       crypto.PrivKey `json:"priv_key"`
-
-	// For persistence.
-	// Overloaded for testing.
+	Key           FilePVKey
```

Comment:
```
Key           FilePVKey     `json:"key"`
LastSignState FilePVLastSignState `json:"last_sign_state"`
```

Reply:
it seems we don't need to transform this struct to json format.